lms/lib/courseware_search/lms_filter_generator.py

```diff
@@ -3,13 +3,35 @@
     * Filter by all courses in which the user is enrolled in
 """
 from student.models import CourseEnrollment
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
 
 from search.filter_generator import SearchFilterGenerator
+from openedx.core.djangoapps.course_groups.cohorts import get_cohort
+from openedx.core.djangoapps.course_groups.models import CourseUserGroupPartitionGroup
 
 
 class LmsSearchFilterGenerator(SearchFilterGenerator):
     """ SearchFilterGenerator for LMS Search """
 
+    def filter_dictionary(self, **kwargs):
+        """ base implementation which filters via start_date """
+        filter_dictionary = super(LmsSearchFilterGenerator, self).filter_dictionary(**kwargs)
+        if 'user' in kwargs and 'course_id' in kwargs and kwargs['course_id']:
+            try:
+                course_key = CourseKey.from_string(kwargs['course_id'])
+            except InvalidKeyError:
+                course_key = SlashSeparatedCourseKey.from_deprecated_string(kwargs['course_id'])
+            cohort = get_cohort(kwargs['user'], course_key, assign=False)
```

Comment:
I realized on my walk home that my proposed solution doesn't cover the case of a staff user seeing all content. In the case that the user is staff and is not masquerading as a student, then there should be no content_groups filter added. I think you need to call ``has_access(user, 'staff', course_key)`` to determine if the user has staff access to the course.